qiita_db/portal.py

```diff
@@ -5,20 +5,156 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
+import warnings
 
 from .sql_connection import SQLConnectionHandler
 from .util import convert_to_id
-from qiita_core.qiita_settings import qiita_config
+from .base import QiitaObject
+from .exceptions import QiitaDBError, QiitaDBDuplicateError, QiitaDBWarning
+from qiita_core.exceptions import IncompetentQiitaDeveloperError
 
-# Only make functions available for main portal
-if qiita_config.portal == "QIITA":
-    def get_studies_by_portal(portal):
-        """Returns study id for all Studies belonging to a portal
+
+class Portal(QiitaObject):
+    _table = 'portal_type'
+
+    def __init__(self, portal):
+        self.portal = portal
+        portal_id = convert_to_id(portal, 'portal_type', 'portal')
+        super(Portal, self).__init__(portal_id)
+
+    @classmethod
+    def create(cls, portal, desc):
+        """Creates a new portal on the system
+
+        Parameters
+        ----------
+        portal : str
+            The name of the portal to add
+        desc : str
+            Description of the portal
+
+        Raises
+        ------
+        QiitaDBDuplicateError
+            Portal already exists
+        """
+        if cls.exists(portal):
+            raise QiitaDBDuplicateError("Portal", portal)
+
+        # Add portal and default analyses for all users
+        sql = """DO $do$
+            DECLARE
+                pid bigint;
+                eml varchar;
+                aid bigint;
+            BEGIN
+                INSERT INTO qiita.portal_type (portal, portal_description)
+                VALUES (%s, %s) RETURNING portal_type_id INTO pid;
+
+                FOR eml IN
+                    SELECT email FROM qiita.qiita_user
+                LOOP
+                    INSERT INTO qiita.analysis
+                    (email, name, description, dflt, analysis_status_id)
+                    VALUES (eml, eml || '-dflt', 'dflt', true, 1) RETURNING
+                    analysis_id INTO aid;
+
+                    INSERT INTO qiita.analysis_workflow (analysis_id, step)
+                    VALUES (aid, 2);
+
+                    INSERT INTO qiita.analysis_portal
+                    (analysis_id, portal_type_id)
+                    VALUES (aid, pid);
+                END LOOP;
+            END $do$;"""
+        conn_handler = SQLConnectionHandler()
+        conn_handler.execute(sql, [portal, desc])
+
+        return cls(portal)
+
+    @staticmethod
+    def delete(portal):
+        """Removes a portal from the system
```

Comment:
Removes a portal from the system -> Removes a portal and default analyses from the system.